aws-android-sdk-testutils/src/main/java/com/amazonaws/testutils/AWSTestBase.java

```diff
@@ -132,103 +121,81 @@ public String getAccountId() {
+            String fileContents = readRawResourceContents(resourceName);
+            mJSONConfiguration  = new JSONConfiguration(new JSONObject(fileContents));
+            return mJSONConfiguration;
+        } catch (JSONException configurationFileError) {
+            throw new RuntimeException(
+                "Failed to read " + TEST_CONFIGURATION_FILENAME + " please check that it is correctly formed.",
+                configurationFileError
+            );
+        }
+    }
 
-    public static JSONConfiguration getJSONConfiguration()
-    {
-        Context context = getContext() ;
-        if (mJSONConfiguration == null) {
-            try {
-                Resources resources = context.getResources();
-                String packageName = context.getPackageName();
-                @RawRes int resourceId =
-                    resources.getIdentifier(TEST_CONFIGURATION_FILENAME, "raw", packageName);
-                InputStream inputStream = resources.openRawResource(resourceId);
-                Scanner in = new Scanner(inputStream);
-                StringBuilder sb = new StringBuilder();
-                while (in.hasNextLine()) {
-                    sb.append(in.nextLine());
-                }
-                in.close();
-
-                mJSONConfiguration  = new JSONConfiguration(new JSONObject(sb.toString()));
-            } catch (Exception je) {
-                throw new RuntimeException(
-                        "Failed to read " + TEST_CONFIGURATION_FILENAME + " please check that it is correctly formed.",
-                        je);
-            }
+    @SuppressWarnings("SameParameterValue")
+    private static String readRawResourceContents(String rawResourceName) {
+        Context context = InstrumentationRegistry.getContext();
+        Resources resources = context.getResources();
+        String packageName = context.getPackageName();
+        @RawRes int resourceId =
+            resources.getIdentifier(rawResourceName, "raw", packageName);
+        InputStream inputStream = resources.openRawResource(resourceId);
+        Scanner in = new Scanner(inputStream);
+        StringBuilder sb = new StringBuilder();
+        while (in.hasNextLine()) {
+            sb.append(in.nextLine());
         }
-        return mJSONConfiguration;
+        in.close();
+        return sb.toString();
     }
+
     public static String getAccessKey() {
         return getJSONConfiguration().getAccessKey();
     }
-    public static String  getSecretKey() {
+
+    public static String getSecretKey() {
         return getJSONConfiguration().getSecretKey();
     }
+
+    public static String getSessionToken() {
+        return getJSONConfiguration().getSessionToken();
+    }
+
     public static String getAccountId() {
         return getJSONConfiguration().getAccountId();
     }
+
     public static JSONObject  getPackageConfigure(String packageName) {
```

Comment:
What is the structure of testconfiguration.json file? Is it like this?
`{ accessKey: "", 
    secretKey: "",
    sessionToken: "",
    Packages: {
         <package ID> : {
              <resource ID> : <resource value>
         }
    }
}`
To keep things consistent across platforms, it's good to have a common "package ID" and a strategy to name "resource ID"

Reply:
Hey @phani-srikar -- the existing structure is like:
```
{
  "Credentials": {
    "accessKey": "asdfasdf",
    "secretKey": "asdfasdf",
    "sessionToken": "asdfasdf",
    "accountId": "asdfasdf"
  },
  "Packages": {
    "iot": {
       ...
    }
  }
}
```
I just added the `"sessionToken"` for this PR. 

I believe we do have the package ID and resource ID keys already.

Reply:
do we need "region" key too in the "credentials" section which would act as default region for all tests unless an override is specified for a package.